docs/servo_power_test/src/config.ts

```diff
@@ -4,17 +4,41 @@ import {ServoController} from './servo_controller';
 import {TestRunner} from './test_runner';
 import {Ui} from './ui';
 
+export class IterationData {
+  public powerDataList: Array<PowerData>;
+  public annotationList: AnnotationDataList;
+  constructor(
+    powerDataList: Array<PowerData>,
+    annotationList: AnnotationDataList
+  ) {
+    this.powerDataList = powerDataList;
+    this.annotationList = annotationList;
+  }
+  public extractData(marginTime: number) {
+    const extractedData = [];
+    for (const powerData of this.powerDataList) {
+      if (
+        this.annotationList.get('start')! + marginTime <= powerData[0] &&
+        powerData[0] <= this.annotationList.get('end')! - marginTime
+      ) {
+        extractedData.push(powerData[1]);
+      }
+    }
+    return extractedData;
+  }
+}
+
 export class Config {
   private INTERVAL_MS = 100;
   private ui: Ui;
   private servoController: ServoController;
   private runner: TestRunner;
   private halt = true;
   private inProgress = false;
-  public powerDataList: Array<PowerData> = [];
-  public annotationList: AnnotationDataList = new Map<string, number>();
+  private iterationDataList: Array<IterationData> = [];
   public graph: Graph;
   public customScript: string;
+  public currentItrNum = 0;
```

Comment:
Could you make this private...? (without an explicit reason, all the class properties should be private.)